### PR TITLE
Hp image index

### LINKED ISSUE #3: Image Index
#### As a user I want the homepage to display all saved images.

__Story__:

Now that we are saving images for our users, we want them to be able to view
the list of images in the system. You might be thinking, "what if my users only
want to view images of cats", or "what if my users only want to see the images
they uploaded". Those are great questions. Maybe you will have the opportunity
to add those features in a future story. For now, we will keep it simple and
quickly deliver a little more value to our users.

__Acceptance criteria__:
- [x] New images that are added show up on the homepage.
- [x] These images are persisted if the browser is closed or even if the
  server is restarted.
- [x] Images are not displayed wider than 400px.
- [x] Newest images appear first.

__Dependencies__:
- Save Image Link

### PR DESCRIPTION
**Pull request by Ads7**
Closes #3

---

The following is a checklist to remind you of the steps to take when opening a
new PR. The horizontal line above and everything below can be removed from the
pull request description.

- [x] I have replaced `X` above with the associated issue number.
- [x] I have changed the issue's label to "pull request".
- [x] I have assigned the designated reviewer to this PR.
- [x] I have set the appropriate label on this PR.
- [ ] I have written clear commit messages ([ref](http://chris.beams.io/posts/git-commit/)).

If needed, please review the document on the
[Image Sharing Project](https://sites.google.com/a/appfolio.com/eng/new-hire-resources/engineering-academy/ropes/image-sharing-project)
for information on how labels should be used.

### CODE REVIEW

File: app/controllers/images_controller.rb [hunk 1]
```diff
@@ -23,3 +24,4 @@ def image_params
     params.require(:image).permit(:url)
   end
 end
+œœ
```
Comment (HaominZhang) [inline app/controllers/images_controller.rb]: Should we delete this?

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,5 +1,7 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
 
-  def home; end
+  def home
```
Comment (HaominZhang) [inline app/controllers/application_controller.rb]: I think there is nothing wrong putting the logic here, but if we think about from the resource perspective, I think it is better to have an `index` action inside `images_controller` and display all images there. Then we can also change our root url to images#index so we can completely have a blank application controller since it doesn't attach to any resources. How do you think?

Comment (Ads7) [reply app/controllers/application_controller.rb]: thought of that to just that might be creating hard link between base page and images. In this case that approach can provide some cleaning too.

File: test/controllers/images_controller_test.rb [hunk 3]
```diff
@@ -1,7 +1,44 @@
...
+    assert_select '.card-title', @image.id.to_s
+  end
+
+  def test_show
+    get image_path(@image.id)
+
+    assert_response :ok
+    assert_select '#image-url', @image.url
+  end
+
```
Comment (RitamBanerjee) [inline test/controllers/images_controller_test.rb]: Good tests! Some of these tests, however, probably belong in a different commit. Specifically the commit in which the method they are testing was added. Ok to leave it for now :)

File: test/models/image_test.rb [hunk 4]
```diff
@@ -12,7 +12,7 @@ class ImageTest < ActiveSupport::TestCase
   end
 
   test 'create valid image' do
-    IMAGE_URL = 'https://images.pexels.com/photos/414612/pexels-photo-414612.jpeg'.freeze
+    IMAGE_URL = 'https://test.jpeg'.freeze
```
Comment (RitamBanerjee) [inline test/models/image_test.rb]: good! Good to be generic in tests for readability purposes
